chalk-solve/src/clauses/program_clauses.rs

```diff
@@ -337,6 +337,63 @@ impl<I: Interner> ToProgramClauses<I> for StructDatum<I> {
     }
 }
 
+impl<I: Interner> ToProgramClauses<I> for FnDefDatum<I> {
+    fn to_program_clauses(&self, builder: &mut ClauseBuilder<'_, I>) {
+        debug_heading!("FnDatum::to_program_clauses(self={:?})", self);
+
+        let interner = builder.interner();
+        let binders = self.binders.map_ref(|b| &b.where_clauses);
+        builder.push_binders(&binders, |builder, where_clauses| {
+            let self_appl_ty = &ApplicationTy {
+                name: self.id.cast(interner),
+                substitution: builder.substitution_in_scope(),
+            };
+            let self_ty = self_appl_ty.clone().intern(interner);
+
+            // forall<T> {
+            //     WF(foo<T>) :- WF(T: Eq).
```

Comment:
This comment kind of makes no sense without a bit more context :)

Reply:
It's intended to refer into an example that would be at the top of the function. For example

```rust
fn foo<T: Eq>() 
```